src/test/java/com/jcabi/s3/mock/MkOcketTest.java

```diff
@@ -0,0 +1,105 @@
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.s3.mock;
+
+import com.amazonaws.services.s3.Headers;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.jcabi.s3.Bucket;
+import com.jcabi.s3.Ocket;
+import com.jcabi.s3.Region;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+/**
+ * Test case for {@link MkOcket}.
+ *
+ * @author Piotr Pradzynski (prondzyn@gmail.com)
+ * @version $Id$
+ * @since 0.10.1
+ */
+public final class MkOcketTest {
+
+    /**
+     * Temp directory.
+     * @checkstyle VisibilityModifierCheck (5 lines)
+     */
+    @Rule
+    public final transient TemporaryFolder temp = new TemporaryFolder();
+
+    /**
+     * Commonly used bucket.
+     */
+    private transient Bucket bucket;
+
+    /**
+     * Ocket to write in it.
+     */
+    private transient Ocket write;
+
+    /**
+     * Sets up common part of tests.
+     * @throws Exception If fails
+     */
+    @Before
+    public void setUp() throws Exception {
+        final Region region = new MkRegion(this.temp.newFolder());
+        this.bucket = region.bucket("test");
+        this.write = this.bucket.ocket("hello.txt");
+    }
+
+    /**
+     * Content Type and Length can be read from MkOcket metadata.
+     * @throws Exception If fails
+     */
+    @Test
+    public void readContentTypeAndLengthFromMetadata() throws Exception {
+        final String text = "hello, world!";
+        new Ocket.Text(this.write).write(text);
+        final Ocket read = new Ocket.Text(this.bucket.ocket(this.write.key()));
+        final ObjectMetadata metadata = read.meta();
+        Assert.assertEquals("text/plain", metadata.getContentType());
+        Assert.assertEquals(text.length(), metadata.getContentLength());
+    }
+
+    /**
+     * Date can be read from MkOcket metadata.
+     * @throws Exception If fails
+     */
+    @Test
+    public void readDateFromMetadata() throws Exception {
+        final Ocket read = new Ocket.Text(this.bucket.ocket(this.write.key()));
+        final ObjectMetadata metadata = read.meta();
```

Comment:
@prondzyn there is a chance qulice complains (not sure if that bug was resolved it) so you might have to do this at the end - yeah, it looks weird :)
```java
final ObjectMetadata metadata = new Ocket.Text(
   this.bucket.ocket(this.write.key())
)
    .meta();
```